mobile/src/fingerprint-popup/fingerprint-popup.android.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { Component } from 'react';
+import FingerprintScanner from 'react-native-fingerprint-scanner';
+import { FingerprintPopupStyle } from './fingerprint-popup.style';
+import { Image, StyleProp, Text, TextStyle, TouchableOpacity, View, ViewStyle } from 'react-native';
+import Modal from 'react-native-modal';
+
+//============================================================================
+interface FingerprintPopupState {
+    error: boolean;
+}
+
+//============================================================================
+interface FingerprintPopupProps {
+    isVisible: boolean;
+    onPopupClose: () => void;
+    onPopupHidden: (success: boolean) => void;
+}
+
+//============================================================================
+export class FingerprintPopupAndroid extends Component<FingerprintPopupProps, FingerprintPopupState> {
+    private success = false;
+
+    //----------------------------------------------------------------------------
+    constructor(props: FingerprintPopupProps) {
+        super(props);
+
+        this.handleAuthenticationAttempt = this.handleAuthenticationAttempt.bind(this);
+        this.fingerprintContainerColor = this.fingerprintContainerColor.bind(this);
+        this.descriptionColor = this.descriptionColor.bind(this);
+        this.descriptionText = this.descriptionText.bind(this);
+
+        this.state = {
+            error: false,
+        };
+    }
+
+    //----------------------------------------------------------------------------
+    public componentDidMount(): void {
+        FingerprintScanner.authenticate({ onAttempt: this.handleAuthenticationAttempt })
+            .then(() => {
+                this.success = true;
+                this.props.onPopupClose();
+            })
+            .catch((error) => {
+                this.success = false;
+                this.props.onPopupClose();
+            });
+    }
+
+    //----------------------------------------------------------------------------
+    public componentWillUnmount(): void {
+        FingerprintScanner.release();
+    }
+
+    //----------------------------------------------------------------------------
+    public render() {
+        return (
+            <Modal
+                isVisible={this.props.isVisible}
+                style={{ justifyContent: 'flex-end', margin: 0 }}
+                onModalHide={() => { this.props.onPopupHidden(this.success); }}>
+                <View style={FingerprintPopupStyle.view.container}>
+                    <Text style={FingerprintPopupStyle.text.title}>{'Sign in'}</Text>
+                    <View style={[FingerprintPopupStyle.view.fingerImageContainer, this.fingerprintContainerColor()]}>
+                        <Image style={FingerprintPopupStyle.view.fingerImage}
+                               source={require('./fingerprint_white_192x192.png')}/>
```

Comment:
move to somewhere above
const fingerPrintImage = require('./fingerprint_white_192x192.png')

Reply:
the reason both here and in `onModalHide` is that it returns new reference every `render` call causing view to rerender

Reply:
I see. Addressed in 29b70b8f7989b0d930725c2c79041cf00c4c8d12